.github/workflows/apps/new_latest_major_versions.go

```diff
@@ -0,0 +1,307 @@
+		return "", fmt.Errorf("failed to parse JSON output: %w", err)
+	}
+
+	// Check if the Origin.URL field exists
+	if moduleInfo.Origin.URL != "" {
+		return moduleInfo.Origin.URL, nil
+	}
+
+	// If Origin.URL is not found, return an error
+	return "", fmt.Errorf("Origin.URL not found in JSON for %s@%s", repository, version)
+}
+
+// Run `git ls-remote` and fetch tags
+func getTags(vcsURL string) ([]string, error) {
+	cmd := exec.Command("git", "ls-remote", "--tags", vcsURL)
+	output, err := cmd.Output()
+	if err != nil {
+		return nil, err
+	}
+	scanner := bufio.NewScanner(bytes.NewReader(output))
+	tags := []string{}
+	tagRegex := regexp.MustCompile(`refs/tags/(v[\d.]+)$`)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if matches := tagRegex.FindStringSubmatch(line); matches != nil {
+			tags = append(tags, matches[1])
+		}
+	}
+	return tags, nil
+}
+
+// Group tags by major version
+func groupByMajor(tags []string) map[string][]string {
+	majors := make(map[string][]string)
+	for _, tag := range tags {
+		if semver.IsValid(tag) {
+			major := semver.Major(tag)
+			majors[major] = append(majors[major], tag)
+		}
+	}
+	return majors
+}
+
+// Get the latest version for a list of versions
+func getLatestVersion(versions []string) string {
+	sort.Slice(versions, func(i, j int) bool {
+		return semver.Compare(versions[i], versions[j]) < 0
+	})
+	return versions[len(versions)-1]
+}
+
+func fetchGoMod(url string) (bool, string) {
+	cmd := exec.Command("curl", "-v", "-s", "-f", url)
+	var stdout, stderr bytes.Buffer
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+
+	err := cmd.Run()
+	if err != nil {
+		// If `curl` fails, check if it's because of a 404
+		if exitErr, ok := err.(*exec.ExitError); ok && exitErr.ExitCode() == 22 { // HTTP 404
+			return false, ""
+		}
+		fmt.Printf("Error running curl: %v\n", err)
+		return false, ""
+	}
+
+	// Parse the output to check for a `module` line
+	scanner := bufio.NewScanner(&stdout)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "module ") {
+			return true, strings.TrimSpace(strings.TrimPrefix(line, "module "))
+		}
+	}
+
+	return false, ""
+}
+
+func truncateVersion(pkg string) string {
+	// Regular expression to match ".v{X}" or "/v{X}" where {X} is a number
+	re := regexp.MustCompile(`(\.v\d+|/v\d+)$`)
+	// Replace the matched pattern with an empty string
+	return re.ReplaceAllString(pkg, "")
+}
+
+// split, ex v5.3.2 => v5
+func truncateMajorVersion(version string) string {
+	parts := strings.Split(version, ".")
+	return parts[0]
+}
+
+// Comparator for version strings
+func compareVersions(v1, v2 string) bool {
+	// if v1 > v2 return true
+	re := regexp.MustCompile(`^v(\d+)$`)
+
+	// Extract the numeric part from the first version
+	match1 := re.FindStringSubmatch(v1)
+	match2 := re.FindStringSubmatch(v2)
+
+	if len(match1) < 2 || len(match2) < 2 {
+		panic("Invalid version format") // Ensure valid versions like "v1", "v2", etc.
```

Comment:
My understanding of `panic` is that will be fatal for this script 

What about logging / returning false here instead so that it can continue? (or maybe this would be moot with using `semver.Compare`